### DISCUSSION TITLE
compoments duplicate / reuse improvement

### DISCUSSION THREAD

**Discussion by samg666** [Budibase/budibase#4670] [Ideas]
Hello,

Many of my pages has similiar repeater, filters, sorting and tabs as below screenshot, currently I can copy/past to duplicate compoments, but the problem is duplicate will just copy the layout but without Javascript codes and binding behind the buttons/data provider/repeater, and we have to copy/past the code one-by-one manually. 

My question is, is there any easier way to reuse a compoments? Or is it possible that copy/past can keep the codes and bindings, then we just need to make minor change to quickly reuse a similar group of compoments. 

<img width="2563" alt="image" src="https://user-images.githubusercontent.com/29649651/155375184-8d5957fc-e7ca-4194-a985-2c8a8b75c095.png">

**Comment by mjashanks** [comment]
Hey Sam. There is no easy way at the moment. 

However, we do plan to add a "reusable components" feature, whereby you can build up your own components and save them for re-use. We're not going to have this in the next few months - but it will come.

Sorry I can't help you more with this!

**Reply by samg666** [reply]
Thanks Michael. Reusable feature is great but seems takes times. The current copy/past is good, the only issue is it just copied the layout but didn't included the codes and binding, will you consider to improve the copy/past to keep the codes and bindings?

**Reply by mjashanks** [reply]
There's a chance that we might make the copy/paste work in our March release: #3818 - if we get to it!
